File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.py
import math

def solution(arr):
    answer = 0
    gndstack = []
    
    for i in range(len(arr)) :
        if i == 0 :
            gndstack.append(arr[i])
            continue
        else:
            compare = gndstack.pop()
            gndstack.append( (compare * arr[i]) // int(math.gcd(compare,arr[i])) )
            
    answer = gndstack[-1]
    return answer

File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.java
class Solution {
    public int solution(int[] arr) {
        int answer = arr[0];
        for (int i = 1 ; i <arr.length;i++)
        {
            int gcd = gcd(answer,arr[i]);
            answer = (answer*arr[i])/gcd;
        }
        return answer;
    }
    
    public int gcd(int a, int b)
    {
        while (b>0)
        {
            int c = b;
            b= a %b;
            a = c;
        }
        
        return a;
    }
}